### ISSUE TITLE
Mamification fails when using react-native-vision-camera on Android

### ISSUE THREAD

**Issue by nickplucker** [msintuneappsdk/ms-intune-app-sdk-android#153]
**Describe the bug:**
We use [react-native-vision-camera](https://github.com/mrousavy/react-native-vision-camera) along with [vision-camera-code-scanner](https://github.com/rodgomesc/vision-camera-code-scanner) to scan barcodes.
Our Android builds use the integrated Intune SDK and we get the following error on the `transformClassesWithMamifyTransformForDebug` task:

```
Execution failed for task ':app:transformClassesWithMamifyTransformForDebug'.
> class com.google.android.gms.common.api.internal.GoogleApiManager$zaa can not implement com.google.android.gms.common.api.internal.zar, because it is not an interface (com.google.android.gms.common.api.internal.zar is in unnamed module of loader javassist
```

**_Our builds work fine without the Intune SDK, which is the reason for making this issue here._** 

**To Reproduce**
Install the latest of [react-native-vision-camera](https://github.com/mrousavy/react-native-vision-camera) along with [vision-camera-code-scanner](https://github.com/rodgomesc/vision-camera-code-scanner) on any React Native project with the Intune SDK. 

Notable libraries used: 
* React Native 68.5
* `com.android.tools.build:gradle:7.1.2`
* `com.google.gms:google-services:4.3.13`
* `org.jetbrains.kotlin:kotlin-gradle-plugin:1.6.21`
* Intune SDK 9.1.0

**Expected behavior:**
Android builds to succeed. 

**Screenshots and logs:**
```
Execution failed for task ':app:transformClassesWithMamifyTransformForDebug'.
> class com.google.android.gms.common.api.internal.GoogleApiManager$zaa can not implement com.google.android.gms.common.api.internal.zar, because it is not an interface (com.google.android.gms.common.api.internal.zar is in unnamed module of loader javassist
```
**Smartphone (please complete the following information):**
 Any Android phone 

**Intune App SDK for Android (please complete the following information):**
- What version of the Intune SDK are you using? Are you using the latest version? **Latest (9.1.0)**
- What platform is your app based in (Java, Xamarin based, Cordova, etc)? **React Native**
- For errors during build, does the app build without Intune SDK integration? **Yes**
- For errors post build, does the app launch without being Intune SDK integrated? **Yes**
- Who is the customer? **Omitted** 
- Do you see a trend with it only being reproduced on a specific device? **No, all Android devices affected**

**Additional context:**
I'm sure there's some sort of dependency conflict, but can't pinpoint where. 
I've tried changing the versions of quite a few libraries, but always end up with the same error.

**Comment by ianbby** [comment]
Hi Intune team, adding additional context here -- this error is _only_ present when we have verify=true in the intunemam gradle settings block. (re: https://learn.microsoft.com/en-us/mem/intune/developer/app-sdk-android-phase3#verification )

Is verification error a risk?

**Comment by mcsimons** [comment]
Hi @nickplucker has this task ever worked? i.e. with an older Intune SDK or older releases of react-native-vision-camera or vision-camera-code-scanner? Or are these new additions to your application?

**Comment by nickplucker** [comment]
Hi @mcsimons, 
This is a new addition to our application, however we just recently transitioned from Intune wrapping to using the SDK.

Those two packages work with Intune wrapping and the SDK (if we turn verification off). As @ianbby mentioned, we only get this error with the verification task. 

I've tried different versions of the SDK and vision-camera, but always get the same error. 

As a workaround, we can disable verification, however, is this approach discouraged?

**Comment by bannus** [comment]
Hi @nickplucker, sorry for the delayed response here. Continuing with verification disabled is discouraged, as the verification failure indicates that an issue may occur when this class is used at runtime. We will try to reproduce on our end -- if you have a minimal sample app that can reproduce the issue, please share it here to aid our investigation.

##### (Internal tracking: #21257934)

**Comment by codylund** [comment]
Unfortunately, I'm not having much luck with a local repro, so I think a minimal sample project could be helpful here.

I will say, I'm not immediately convinced that this is specific to React or the `react-native-vision-camera` and `vision-camera-code-scanner` libraries, because the conflicting classes appear to originate from the Google Play Services libraries. I agree with the original assessment that there may be some version dependency conflicts in the Play Services dependencies. @nickplucker, are you able to provide any additional context on how Google Play Services are configured for your application? I'd be particularly interested to see the corresponding Gradle logic.

With no certainty, you may also have success using the preview version of our plug-in, available in v9.4.0 of the MAM SDK, as documented [here](https://github.com/msintuneappsdk/ms-intune-app-sdk-android/blob/master/CHANGELOG.md#version-940). The preview version does not utilize the now-deprecated Gradle transform APIs, which could be at play here.

**Comment by bannus** [comment]
Closing due to lack of response. Will re-open if more details are provided.
